app-cdr/cdrtools/cdrtools-3.01_alpha17-r1.ebuild

```diff
@@ -0,0 +1,172 @@
+)
+
+FILECAPS=(
+	cap_sys_resource,cap_dac_override,cap_sys_admin,cap_sys_nice,cap_net_bind_service,cap_ipc_lock,cap_sys_rawio+ep usr/bin/cdrecord --
+	cap_dac_override,cap_sys_admin,cap_sys_nice,cap_net_bind_service,cap_sys_rawio+ep usr/bin/cdda2wav --
+	cap_dac_override,cap_sys_admin,cap_net_bind_service,cap_sys_rawio+ep usr/bin/readcd
+)
+
+src_prepare() {
+	default
+
+	gnuconfig_update
+
+	# Remove profiled make files.
+	find -name '*_p.mk' -delete
+
+	# Adjusting hardcoded paths.
+	sed -i -e 's:opt/schily:usr:' \
+		$(find ./ -type f -name \*.[0-9ch] -exec grep -l 'opt/schily' '{}' '+') \
+		|| die "sed opt/schily"
+
+	sed -i -e "s:\(^INSDIR=\t\tshare/doc/\):\1${PF}/:" \
+		$(find ./ -type f -exec grep -l 'INSDIR.\+doc' '{}' '+') \
+		|| die "sed doc"
+
+	# Respect libdir.
+	sed -i -e "s:\(^INSDIR=\t\t\)lib:\1$(get_libdir):" \
+		$(find ./ -type f -exec grep -l '^INSDIR.\+lib\(/siconv\)\?$' '{}' '+') \
+		|| die "sed multilib"
+
+	# Do not install static libraries.
+	sed -i -e 's:include\t\t.*rules.lib::' \
+		$(find ./ -type f -exec grep -l '^include.\+rules\.lib' '{}' '+') \
+		|| die "sed rules"
+
+	# Respect CC/CXX variables.
+	cd "${S}"/RULES
+	local tcCC=$(tc-getCC)
+	local tcCXX=$(tc-getCXX)
+	sed -i -e "/cc-config.sh/s|\$(C_ARCH:%64=%) \$(CCOM_DEF)|${tcCC} ${tcCC}|" \
+		rules1.top || die "sed rules1.top"
+	sed -i -e "/^\(CC\|DYNLD\|LDCC\|MKDEP\)/s|gcc|${tcCC}|" \
+		-e "/^\(CC++\|DYNLDC++\|LDCC++\|MKC++DEP\)/s|g++|${tcCXX}|" \
+		-e "/COPTOPT=/s/-O//" \
+		-e 's/$(GCCOPTOPT)//' \
+		cc-gcc.rul || die "sed cc-gcc.rul"
+	sed -i -e "s|^#CONFFLAGS +=\t-cc=\$(XCC_COM)$|CONFFLAGS +=\t-cc=${tcCC}|g" \
+		rules.cnf || die "sed rules.cnf"
+
+	# Create additional symlinks needed for some archs (armv4l already created)
+	local t
+	for t in armv4tl armv5tel armv7l ppc64 s390x; do
+		ln -s i586-linux-cc.rul ${t}-linux-cc.rul || die
+		ln -s i586-linux-gcc.rul ${t}-linux-gcc.rul || die
+	done
+
+	# Schily make setup.
+	cd "${S}"/DEFAULTS
+	local os="linux"
+	[[ ${CHOST} == *-darwin* ]] && os="mac-os10"
+
+	sed -i \
+		-e "s:/opt/schily:/usr:g" \
+		-e "s:/usr/src/linux/include::g" \
+		-e "s:bin:root:g" \
+		-e '/^DEFUMASK/s,002,022,g' \
+		Defaults.${os} || die "sed Schily make setup"
+	# re DEFUMASK above:
+	# bug 486680: grsec TPE will block the exec if the directory is
+	# group-writable. This is painful with cdrtools, because it makes a bunch of
+	# group-writable directories during build. Change the umask on their
+	# creation to prevent this.
+}
+
+# skip obsolete configure script
+src_configure() { : ; }
+
+src_compile() {
+	if use unicode; then
+		local flags="$(test-flags -finput-charset=ISO-8859-1 -fexec-charset=UTF-8)"
+		if [[ -n ${flags} ]]; then
+			append-flags ${flags}
+		else
+			ewarn "Your compiler does not support the options required to build"
+			ewarn "cdrtools with unicode in USE. unicode flag will be ignored."
+		fi
+	fi
+
+	if ! use caps; then
+		CFLAGS="${CFLAGS} -DNO_LINUX_CAPS"
```

Comment:
Seeing that you're passing `$CPPFLAGS` to emake, you might as well pass all macro options (`-Dblabla`) via `CPPFLAGS` too. And if you're adding stuff to FLAGS, please use `flag-o-matic` instead of adding them yourself.